VersionMiner/Services/IHttpClient.cs

```diff
@@ -0,0 +1,127 @@
+// <copyright file="IHttpClient.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+using System.Collections.ObjectModel;
+
+namespace VersionMiner.Services;
+
+/// <summary>
+/// Provides a class for sending HTTP requests and receiving HTTP responses from a resource identified by a URI.
+/// </summary>
+public interface IHttpClient : IDisposable
+{
+    /// <summary>
+    /// Gets or sets the BaseUrl property for requests made by this client instance.
+    /// </summary>
+    string BaseUrl { get; set; }
+
+    /// <summary>
+    /// Gets the list of the headers for the next request that will be executed.
+    /// </summary>
+    ReadOnlyCollection<(string name, string value)> RequestHeaders { get; }
+
+    /// <summary>
+    /// Gets the list of headers that are made with every request.
+    /// </summary>
+    ReadOnlyCollection<(string name, string value)> DefaultHeaders { get; }
+
+    /// <summary>
+    /// Adds a header to the next request using the given <paramref name="name"/> and <paramref name="value"/>.
+    /// </summary>
+    /// <param name="name">The name of the header to add.</param>
+    /// <param name="value">The value of the header to add.</param>
+    /// <remarks>
+    ///     After the next request is made, the headers will be cleared out.
```

Comment:
Grammar - can't end sentence w/ a preposition
```suggestion
    ///     After the next request is made, the headers will be cleared.
```